components/brave_rewards/browser/rewards_service.h

```diff
@@ -312,6 +312,8 @@ class RewardsService : public KeyedService {
 
   virtual void GetExternalWallet(GetExternalWalletCallback callback) = 0;
 
+  virtual std::string GetExternalWalletType() const = 0;
```

Comment:
Code that relies on this method may need to be rewritten when we improve how we handle external wallets internally.

Currently, `SetExternalWalletType` is used to set a pref, which mostly serves to control which wallet type will be returned from `GetExternalWallet`. This was a hack to work around the fact that the front-end design expects to be able to retrieve a populated `mojom::ExternalWallet` structure, even when the user is not connected to any wallet provider.

Ideally, this pref would not be necessary: the engine would know which wallet is currently connected (if any), and the front-end would not expect to get "login" URLs from the `ExternalWallet` structure. In that case we'd remove the `SetExternalWalletType` method.

Code that relies on `GetExternalWalletType` (a sync method), may have to be rewritten in the future, and I would prefer that we not have more implementations using the `SetExternalWalletType` hack going forward.

Reply:
@zenparsing mentioned in our meeting that this is good for now, but let's not use `GetExternalWalletType` beyond this going forward.